[addons][binary] First automation support part

AlwinEsch opened the pull request:
## Description

This brings in the first few parts of the automated kodi-dev-kit.

Commit 1:
-----------------------
**[addons][cmake] Clean up addon header CMakeLists.txt**

Previously, "core_add_library" was used which, once referring to library by name, and on the other hand always had the path as "addons_kodi-dev-kit_include_ *" in it.

This adds a new CMake function with which it is done via "core_add_devkit_header". The difference is that this checks on "ENABLE_STATIC_LIBS" and always adds "addons_kodi-dev-kit_include_" as begin.

The following background is behind it:
- So that the kodi-dev-kit path is not set in its parts (here CMakeLists.txt) and the folder can simply be moved to another location.
- The "core_add_devkit_header" could possibly be replaced with another function if kodi-dev-kit were used independently of Kodi.
- Better to use with auto scripts

The other `if (HEADERS)` has the background if inline cmake calls were integrated into it and HEADERS would be given as empty.

Commit 2:
-----------------------
**[addons][binary] add initial autogen script parts**

This introduces the first part of the automated dev kit API update system.

The background is to concentrate the manual work in the end only on places which really need hand changes and to do everything else automatically with the script.

When the larger system comes in, it is extremely helpful by means of a script, because a lot of changes of the same kind come in when changes are made and then become tedious by hand.

This smaller commit comes to present my planned style to a manageable extent and hope is acceptable.

Commit 3:
-----------------------
**[auto][addons] devkit script update (21/10/2021 03:27:18)**

See his text on commit there :smile:, is automatic generated by script.

## Motivation and context
Power in the planned revised system, most parts automatically, and by hand can not only refer to desired primary changes.

<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## How has this been tested?
Script calls and Kodi build.
<!--- Please describe in detail how you tested your change. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## What is the effect on users?
<!--- Summarize the effect of this change on Kodi end-users. -->
<!--- If the PR does not have a noticeable impact (e.g., if it only changes documentation), -->
<!--- just leave it empty. Put in more detail the bigger the impact is. -->
<!--- This section may be used for automatic creation of release notes. -->

## Screenshots (if appropriate):
Console output from script:
![Bildschirmfoto vom 2021-10-21 03-30-07](https://user-images.githubusercontent.com/6879739/138196916-b1425f4b-83e1-4bdb-b340-593d56323143.png)

Help return from script:
![Bildschirmfoto vom 2021-10-21 03-32-50](https://user-images.githubusercontent.com/6879739/138196943-15486ca9-63de-4eee-b188-581bd239d7ed.png)


## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

AlwinEsch committed:
[addons][cmake] Clean up addon header CMakeLists.txt

Previously, "core_add_library" was used which, once referring to library by name, and on the other hand always had the path as "addons_kodi-dev-kit_include_ *" in it.

This adds a new CMake function with which it is done via "core_add_devkit_header". The difference is that this checks on "ENABLE_STATIC_LIBS" and always adds "addons_kodi-dev-kit_include_" as begin.

The following background is behind it:
- So that the kodi-dev-kit path is not set in its parts (here CMakeLists.txt) and the folder can simply be moved to another location.
- The "core_add_devkit_header" could possibly be replaced with another function if kodi-dev-kit were used independently of Kodi.
- Better to use with auto scripts

The other `if (HEADERS)` has the background if inline cmake calls were integrated into it and HEADERS would be given as empty.

lrusak reviewed (commented):
Cool!!

I don't have much to say as I can't really do an in depth review but from I do see I think the python stuff would benefit a lot from the use of `f-strings` or `str.format()` rather than simple concatenation. See: https://docs.python.org/3/tutorial/inputoutput.html

AlwinEsch committed:
[addons][binary] add initial autogen script parts

This introduces the first part of the automated dev kit API update system.

The background is to concentrate the manual work in the end only on places which really need hand changes and to do everything else automatically with the script.

When the larger system comes in, it is extremely helpful by means of a script, because a lot of changes of the same kind come in when changes are made and then become tedious by hand.

This smaller commit comes to present my planned style to a manageable extent and hope is acceptable.

AlwinEsch committed:
[auto][addons] devkit script update (26/10/2021 16:26:03)

This commit automatic generated by script 'xbmc/addons/kodi-dev-kit/tools/code-generator.py'.

Used call options: {'force': False, 'debug': False, 'commit': True}
Generate cmake parts...
[26/10/2021 16:26:01] Generate CMakeLists.txt files in xbmc/addons/kodi-dev-kit/include/kodi dirs: See below
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/addon-instance/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/addon-instance/inputstream/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/addon-instance/peripheral/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/addon-instance/pvr/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/c-api/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/c-api/addon-instance/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/c-api/addon-instance/inputstream/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/c-api/addon-instance/pvr/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/c-api/gui/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/c-api/gui/controls/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/c-api/gui/dialogs/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/c-api/gui/input/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/c-api/platform/CMakeLists.txt: Created new
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/c-api/platform/android/CMakeLists.txt:  (Contains limited OS header: android) Created new
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/gui/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/gui/controls/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/gui/dialogs/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/gui/gl/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/gui/input/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/platform/CMakeLists.txt: Created new
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/platform/android/CMakeLists.txt:  (Contains limited OS header: android) Created new
[26/10/2021 16:26:01]  - Check xbmc/addons/kodi-dev-kit/include/kodi/tools/CMakeLists.txt: Updated
[26/10/2021 16:26:01] Check for cmake/treedata/common/addon_dev_kit.txt: Created new
Git update...
[26/10/2021 16:26:01] Perfom GIT update check: See below
[26/10/2021 16:26:01]  - Changed file cmake/treedata/common/addon_dev_kit.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/addon-instance/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/addon-instance/inputstream/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/addon-instance/peripheral/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/addon-instance/pvr/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/c-api/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/c-api/addon-instance/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/c-api/addon-instance/inputstream/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/c-api/addon-instance/pvr/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/c-api/gui/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/c-api/gui/controls/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/c-api/gui/dialogs/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/c-api/gui/input/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/c-api/platform/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/c-api/platform/android/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/gui/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/gui/controls/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/gui/dialogs/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/gui/gl/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/gui/input/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/platform/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/platform/android/CMakeLists.txt: Updated
[26/10/2021 16:26:01]  - Changed file xbmc/addons/kodi-dev-kit/include/kodi/tools/CMakeLists.txt: Updated
[26/10/2021 16:26:01] Perfom GIT commit:
